EnumUtilities/EnumUtilCompiledCache.cs

```diff
@@ -195,6 +195,33 @@ public static partial class EnumUtil<TEnum>
                 .Compile();
         }
 
+        private static Func<TEnum, TEnum> GenerateFlagwiseNot()
+        {
+            var val = Expression.Parameter(typeof(TEnum));
+
+            // Convert from Enum to Enum.'s underlying type (byte, int, long, …)
+            // to allow bitwise functions to work.
+            var valConverted = Expression.Convert(val, Enum.GetUnderlyingType(typeof(TEnum)));
+
+            // ~val
+            var notExpression = Expression.MakeUnary(ExpressionType.Not, valConverted, null);
+
+            // Calculate the value that corresponds to all flags being set.
+            // We can AND this with the bitwise not to restrict the set flags to
+            // actual flags.
+            var allSetValue = default(TEnum);
+            foreach (var flagValue in GetValues()) allSetValue = BitwiseOr(allSetValue, flagValue);
```

Comment:
I didn't know that this would work. As this assumes that during static instantiation BitwiseOr exists and can be accessed when FlagwiseNot is being set.

Feels janky :/

Thoughts?

Reply:
It does need the implementation of `BitwiseOr`. It would seem like a waste to temporarily build an extra `BitwiseOr` just for it. I think I will update `GenerateFlagwiseNot()` to accept an argument which will be an implementation of `BitwiseOr` so that the dependency is visible at the call-site. Then it is obvious that the initialization of `flagwiseNot` has to be after the initialization of `bitwiseOr`.